testing/test_assertrewrite.py

```diff
@@ -117,6 +117,36 @@ def test_dont_rewrite_plugin(self, testdir):
         result = testdir.runpytest_subprocess()
         assert "warnings" not in "".join(result.outlines)
 
+    def test_rewrites_plugin_as_a_package(self, testdir):
+        pkgdir = testdir.mkpydir("plugin")
+        pkgdir.join("__init__.py").write(
+            "import pytest\n"
+            "@pytest.fixture\n"
+            "def special_asserter():\n"
+            "    def special_assert(x, y):\n"
+            "        assert x == y\n"
+            "    return special_assert\n"
+        )
+        testdir.makeconftest('pytest_plugins = ["plugin"]')
+        testdir.makepyfile("def test(special_asserter): special_asserter(1, 2)\n")
+        result = testdir.runpytest()
+        result.stdout.fnmatch_lines(["*assert 1 == 2*"])
+
+    def test_honors_pep_235(self, testdir, monkeypatch):
+        # note: these modules are named `test_*` to trigger rewriting
+        testdir.tmpdir.join("test_y.py").write("x = 1")
```

Comment:
Any reason not to put the file and package directory in the same directory (you had to create a `x` directory and put it into sys.path). If there is, I think it would be nice to leave a comment stating the reason.

Reply:
hmmm they might be able to be in the same entry, let me try that -- I was initially working with `x.py` and `X.py` which definitely need different `sys.path` entries -- but it turned out the bug was with packages anyway! go figure!

Reply:
it didn't work (I couldn't get it to fail on macos with a single path entry) and I honestly don't know the reason why to add a comment :/

Reply:
I would add exactly what you commented:

"# Couldn't get it to fail on macos with a single path entry"

This is good because it clearly states what's going on. If somebody in the future manages to make it work, they can remove the extra path with confidence that the path was there for that reason, not for something else. 😁